cmd/proxy/sctp_proxy.go

```diff
@@ -19,7 +19,12 @@ type SCTPProxy struct {
 
 // NewSCTPProxy creates a new SCTPProxy.
 func NewSCTPProxy(frontendAddr, backendAddr *sctp.SCTPAddr) (*SCTPProxy, error) {
-	listener, err := sctp.ListenSCTP("sctp", frontendAddr)
+	// detect version of hostIP to bind only to correct version
+	ipVersion := ipv4
+	if frontendAddr.IPAddrs[0].IP.To4() == nil {
```

Comment:
@arkodg curious: would this be an issue? https://twitter.com/bradfitz/status/1341979944151207937?s=20

<img width="606" alt="Screenshot 2020-12-24 at 10 36 04" src="https://user-images.githubusercontent.com/1804568/103079295-de01ad80-45d3-11eb-90c5-6bf17e325886.png">


https://github.com/golang/go/issues/37921

Reply:
we currently don't support `IPv4-mapped IPv6 addresses` today, we build out a separate ipv4 and ipv6 stack